docs/06-concepts/18-testing.md

```diff
@@ -0,0 +1,365 @@
+# Testing
+
+Serverpod provides simple but feature rich test tools to make testing your backend a breeze.
+
+## Getting started
+
+Go to the server directory and generate the test tools by running `serverpod generate`. The default location for the generated file is `integration_test/test_tools/serverpod_test_tools.dart`.
+
+The generated file exports a `withServerpod` helper that enables you to call your endpoints directly like regular functions:
+
+```dart
+// Import the generated file, it contains everything you need.
```

Comment:
Since this quote starts with an import it gives the impression it is a complete file. Suggest you make it complete with main() etc as well:

```
import 'package:test/test.dart';
import '../../test_tools/serverpod_test_tools.dart';

void main() {
  withServerpod('Given Example endpoint', (endpoints, session) {
    test('when calling `hello` then should return greeting', () async {
      final greeting = await endpoints.example.hello(session, 'Michael');
      expect(greeting, 'Hello, Michael!');
    });
  });
}

```

Reply:
You're completely right, missed that! 👍

Reply:
Fixed!